tests/test_vision.py

```diff
@@ -40,3 +40,16 @@ def test_normalize(path):
     assert abs(x.std()-1) < abs(m-1)
 
     with pytest.raises(Exception): data.normalize()
+
+def test_denormalize(path):
+    data = ImageDataBunch.from_folder(path, ds_tfms=(rand_pad(2, 28), []))
+    original_x, y = data.valid_dl.one_batch()
+    data.normalize()
+    normalized_x, y = data.valid_dl.one_batch()
+    denormalized = denormalize(normalized_x, original_x.mean(), original_x.std())
+    assert round(original_x.mean().item(), 4) == round(denormalized.mean().item(), 4)
+    assert round(original_x.std().item(), 4) == round(denormalized.std().item(), 4)
+
+def test_training_validation_split(path):
+    data = ImageDataBunch.from_folder(path, ds_tfms=(rand_pad(2, 28), []), valid_pct=0.20)
+    assert len(data.train_ds) > len(data.valid_ds)
```

Comment:
This is too loose, data.train_ds should be 4 times bigger, so let's test for that (maybe a 3.95 or something smaller with the rounding errors).

Reply:
That was my initial thought as well. However, when running this multiple times I found the train/validation ratio to vary widely around 4 (probably because of a smaller sample size) and couldn't account for the difference with rounding errors. Thoughts?

Reply:
That's weird. What's the lower ration you find if you run this, say... 1000 times?

Reply:
Min ~ 3.05
Mean ~ 4.02
Max ~ 5.39